src/NuGet.Core/NuGet.Configuration/Settings/SettingsLoadingContext.cs

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Threading;
+using NuGet.Common;
+
+namespace NuGet.Configuration
+{
+    public sealed class SettingsLoadingContext : IDisposable
+    {
+        private readonly IList<Lazy<SettingsFile>> _settingsFiles = new List<Lazy<SettingsFile>>();
```

Comment:
it doesn't appear to me that the order of items in the list are important. You can reduce the O(n^2) lookup by using a dictionary instead.

Reply:
I don't expect this collection to ever have more than 10 items. 
A dictionary is an overkill.

Reply:
so is writing your own lookup code when you could just use the dictionary's indexer.

Reply:
ListDictionary maybe makes sense. 
A hashtable backed dictionary is definitely not necessary.

Reply:
What's the concern with a hash table backed dictionary, or in what way is it overkill? Memory usage? cpu usage? ListDictionary sounds worse to me than a List<>, with the possible exception of API usage.

You're using _settingsFiles as a dictionary, so it would be fewer lines of code to actually use a dictionary.

Reply:
both memory and CPU usage is worse.
The current approach has no extra allocations in the success case. 

Even on MSDN the guidance is to not use dict for super small collections.